addons/web/static/src/core/clipboard/clipboard_service.js

```diff
@@ -0,0 +1,42 @@
+/** @odoo-module **/
+
+import { registry } from "../registry";
+import { browser } from "@web/core/browser/browser";
+
+/**
+ *  @typedef {{
+ *      writeText(
+ *          String: text,
+ *          options?: ...       TODO
+ *      ): () => void;
+ *  }}
+ */
+
+export const clipboardService = {
+
+    /** @returns {void} */
+    start(env) {
+
+        function writeText(text, options = {}) {
+            setTimeout(async () => { // Use a timeout for the Safari write to clipboard to work
+                try {
+                    await browser.navigator.clipboard.writeText(text);
+                    env.services.notification.add(env._t("Link copied to clipboard."), {type: "success"});
+                    if (options.onSuccess) {
+                        options.onSuccess();
+                    }
+                } catch (error) {
+                    env.services.notification.add(env._t("Error copying to clipboard."), {type: "danger"});
+                    console.error('Error copying to clipboard:', error);
```

Comment:
Don't use a generic catch clause that catches all errors without checking the error type and rethrowing unexpected errors, and keep the try clause as short as possible. With the current code you're catching errors in "onSuccess" which you almost certainly do not want to do.